File: ebean-querybean/src/test/java/org/example/domain/otherpackage/GenericTypeAttributeConverter.java
package org.example.domain.otherpackage;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
public class GenericTypeAttributeConverter implements AttributeConverter<GenericType<GenericTypeArgument>, String> {
  @Override
  public String convertToDatabaseColumn(final GenericType<GenericTypeArgument> attribute) {
    return attribute.getData().name();
  }

  @Override
  public GenericType<GenericTypeArgument> convertToEntityAttribute(final String dbData) {
    return new GenericType<>(GenericTypeArgument.valueOf(dbData));
  }
}

File: tests/test-kotlin/src/main/kotlin/org/example/otherpackage/GenericTypeAttributeConverter.kt
package org.example.otherpackage

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class GenericTypeAttributeConverter : AttributeConverter<GenericType<GenericTypeArgument>, String> {
  override fun convertToDatabaseColumn(attribute: GenericType<GenericTypeArgument>): String {
    return attribute.data.name
  }

  override fun convertToEntityAttribute(dbData: String): GenericType<GenericTypeArgument> {
    return GenericType(GenericTypeArgument.valueOf(dbData))
  }
}
